packages/react-native-reanimated/__tests__/plugin.test.ts

```diff
@@ -147,7 +149,109 @@ describe('babel plugin', () => {
       </script>`;
 
       const { code } = runPlugin(input);
-      expect(code).toContain('const foo=this._recur');
+      expect(code).toMatch(/const foo_null[0-9]+=this._recur;/gm);
+      expect(code).toMatchSnapshot();
+    });
+  });
+
+  describe('names', () => {
+    it('unnamed ArrowFunctionExpression', () => {
+      const input = html`<script>
+        () => {
+          'worklet';
+          return 1;
+        };
+      </script>`;
+
+      const { code } = runPlugin(input);
+      expect(code).toMatch(/function null[0-9]+\(\)/gm);
+      expect(code).toMatchSnapshot();
+    });
+
+    it('unnamed FunctionExpression', () => {
+      const input = html`<script>
+        [
+          function () {
+            'worklet';
+            return 1;
+          },
+        ]();
+      </script>`;
+
+      const { code } = runPlugin(input);
+      expect(code).toMatch(/function null[0-9]+\(\)/gm);
+      expect(code).toMatchSnapshot();
+    });
+
+    it('names ObjectMethod with expression key', () => {
+      const input = html`<script>
+        const obj = {
+          ['foo']() {
+            'worklet';
+          },
+        };
+      </script>`;
+
+      // TODO: this is an edge case that wasn't ever handled.
+      expect(() => runPlugin(input)).toThrow();
+      // const { code } = runPlugin(input);
+      // expect(code).toMatch(/function foo_null[0-9]+\(\)/gm);
+      // expect(code).toMatchSnapshot();
```

Comment:
```suggestion
```

Reply:
I keep it here on purpose since this the correct assertion.